fbw-a380x/src/systems/instruments/src/EWD/shared/ArincValueProvider.ts

```diff
@@ -3,27 +3,87 @@
 // SPDX-License-Identifier: GPL-3.0
 
 import { EventBus } from '@microsoft/msfs-sdk';
-import { Arinc429Register } from '@flybywiresim/fbw-sdk';
+import { Arinc429Register, Arinc429Word } from '@flybywiresim/fbw-sdk';
 import { EwdSimvars } from './EwdSimvarPublisher';
 
 export interface Arinc429Values {
   sat: Arinc429Register;
   slatsFlapsStatus: Arinc429Register;
   slatsPosition: Arinc429Register;
   flapsPosition: Arinc429Register;
+  cpiomB1AgsDiscrete: Arinc429Word;
+  cpiomB2AgsDiscrete: Arinc429Word;
+  cpiomB3AgsDiscrete: Arinc429Word;
+  cpiomB4AgsDiscrete: Arinc429Word;
+  cpiomBAgsDiscrete: Arinc429Word;
 }
 export class ArincValueProvider {
   private sat = Arinc429Register.empty();
 
+  private cpiomB1AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomB2AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomB3AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomB4AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomBToUse = 1;
+
   constructor(private readonly bus: EventBus) {}
 
   public init() {
     const publisher = this.bus.getPublisher<Arinc429Values>();
     const subscriber = this.bus.getSubscriber<EwdSimvars>();
+    this.cpiomBToUse = this.determineCpiomBToUse();
 
     subscriber.on('satRaw').handle((p) => {
       this.sat.set(p);
       publisher.pub('sat', this.sat);
     });
+
+    subscriber.on('cpiomB1AgsDiscreteRaw').handle((discreteWord) => {
+      this.cpiomB1AgsDiscrete = new Arinc429Word(discreteWord);
```

Comment:
**issue:** this will allocate a new object every time `cpiomB1AgsDiscreteRaw` is published, which here seems like it would be every frame.

Instead, use an `Arinc429Register` and call `setWord` on it.

Reply:
Thanks for the suggestion! Done.